test_read_big_table failing

The 'big table' tests for the C reader are currently failing for me - it doesn't look like a critical issue but we should fix the test nonetheless:

```
============================================================================================ FAILURES ============================================================================================
______________________________________________________________________________________ test_read_big_table _______________________________________________________________________________________

cls = <class '_pytest.runner.CallInfo'>, func = <function call_runtest_hook.<locals>.<lambda> at 0x7f93aadb40d0>, when = 'call'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: "Callable[[], TResult]",
        when: "Literal['collect', 'setup', 'call', 'teardown']",
        reraise: Optional[
            Union[Type[BaseException], Tuple[Type[BaseException], ...]]
        ] = None,
    ) -> "CallInfo[TResult]":
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: Optional[TResult] = func()

../../python/dev/lib/python3.9/site-packages/_pytest/runner.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../python/dev/lib/python3.9/site-packages/_pytest/runner.py:255: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
../../python/dev/lib/python3.9/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../../python/dev/lib/python3.9/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../python/dev/lib/python3.9/site-packages/pluggy/manager.py:84: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
../../python/dev/lib/python3.9/site-packages/_pytest/unraisableexception.py:88: in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def unraisable_exception_runtest_hook() -> Generator[None, None, None]:
        with catch_unraisable_exception() as cm:
            yield
            if cm.unraisable:
                if cm.unraisable.err_msg is not None:
                    err_msg = cm.unraisable.err_msg
                else:
                    err_msg = "Exception ignored in"
                msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                msg += "".join(
                    traceback.format_exception(
                        cm.unraisable.exc_type,
                        cm.unraisable.exc_value,
                        cm.unraisable.exc_traceback,
                    )
                )
>               warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
E               
E               Traceback (most recent call last):
E                 File "/home/tom/Code/astropy/astropy/io/ascii/tests/test_c_reader.py", line 1010, in test_read_big_table
E                   assert sum(1 for line in open(filename)) == NB_ROWS + 1
E               ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/pytest-of-tom/pytest-3/test_read_big_table0/big_table.csv' mode='r' encoding='UTF-8'>

../../python/dev/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning
-------------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------------
Creating a 250000 rows table (500 columns).
Saving the table to /tmp/pytest-of-tom/pytest-3/test_read_big_table0/big_table.csv
Counting the number of lines in the csv, it should be 250000 + 1 (header).
Reading the file with astropy.
______________________________________________________________________________________ test_read_big_table2 ______________________________________________________________________________________

cls = <class '_pytest.runner.CallInfo'>, func = <function call_runtest_hook.<locals>.<lambda> at 0x7f93aadb4ca0>, when = 'call'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: "Callable[[], TResult]",
        when: "Literal['collect', 'setup', 'call', 'teardown']",
        reraise: Optional[
            Union[Type[BaseException], Tuple[Type[BaseException], ...]]
        ] = None,
    ) -> "CallInfo[TResult]":
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: Optional[TResult] = func()

../../python/dev/lib/python3.9/site-packages/_pytest/runner.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../python/dev/lib/python3.9/site-packages/_pytest/runner.py:255: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
../../python/dev/lib/python3.9/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../../python/dev/lib/python3.9/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../python/dev/lib/python3.9/site-packages/pluggy/manager.py:84: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
../../python/dev/lib/python3.9/site-packages/_pytest/unraisableexception.py:88: in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def unraisable_exception_runtest_hook() -> Generator[None, None, None]:
        with catch_unraisable_exception() as cm:
            yield
            if cm.unraisable:
                if cm.unraisable.err_msg is not None:
                    err_msg = cm.unraisable.err_msg
                else:
                    err_msg = "Exception ignored in"
                msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                msg += "".join(
                    traceback.format_exception(
                        cm.unraisable.exc_type,
                        cm.unraisable.exc_value,
                        cm.unraisable.exc_traceback,
                    )
                )
>               warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
E               
E               Traceback (most recent call last):
E                 File "/home/tom/Code/astropy/astropy/io/ascii/tests/test_c_reader.py", line 1039, in test_read_big_table2
E                   assert sum(1 for line in open(filename)) == NB_ROWS + 1
E               ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/pytest-of-tom/pytest-3/test_read_big_table20/big_table.csv' mode='r' encoding='UTF-8'>

../../python/dev/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning
-------------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------------
Creating a 214748369 rows table.
Saving the table to /tmp/pytest-of-tom/pytest-3/test_read_big_table20/big_table.csv
Counting the number of lines in the csv, it should be 214748369 + 1 (header).
Reading the file with astropy.
==================================================================================== short test summary info =====================================================================================
FAILED astropy/io/ascii/tests/test_c_reader.py::test_read_big_table - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
FAILED astropy/io/ascii/tests/test_c_reader.py::test_read_big_table2 - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
```

cc @dhomeier @taldcroft

--- a/astropy/io/ascii/tests/test_c_reader.py
+++ b/astropy/io/ascii/tests/test_c_reader.py
@@ -1005,9 +1005,9 @@ def test_read_big_table(tmpdir):
     t.write(filename, format='ascii.csv', overwrite=True)
     t = None
 
-    print("Counting the number of lines in the csv, it should be {}"
-          " + 1 (header).".format(NB_ROWS))
-    assert sum(1 for line in open(filename)) == NB_ROWS + 1
+    print("Counting the number of lines in the csv, it should be {NB_ROWS} + 1 (header).")
+    with open(filename, 'r') as f:
+        assert sum(1 for line in f) == NB_ROWS + 1
 
     print("Reading the file with astropy.")
     t = Table.read(filename, format='ascii.csv', fast_reader=True)
@@ -1023,20 +1023,20 @@ def test_read_big_table2(tmpdir):
     # (2**32 // 2) : max value for int
     # // 10 : we use a value for rows that have 10 chars (1e9)
     # + 5 : add a few lines so the length cannot be stored by an int
-    NB_ROWS = (2**32 // 2) // 10 + 5
+    NB_ROWS = 2**32 // 2 // 10 + 5
     filename = str(tmpdir.join("big_table.csv"))
 
     print(f"Creating a {NB_ROWS} rows table.")
-    data = np.full(2**32 // 2 // 10 + 5, int(1e9), dtype=np.int32)
+    data = np.full(NB_ROWS, int(1e9), dtype=np.int32)
     t = Table(data=[data], names=['a'], copy=False)
 
     print(f"Saving the table to {filename}")
     t.write(filename, format='ascii.csv', overwrite=True)
     t = None
 
-    print("Counting the number of lines in the csv, it should be {}"
-          " + 1 (header).".format(NB_ROWS))
-    assert sum(1 for line in open(filename)) == NB_ROWS + 1
+    print("Counting the number of lines in the csv, it should be {NB_ROWS} + 1 (header).")
+    with open(filename, 'r') as f:
+        assert sum(1 for line in f) == NB_ROWS + 1
 
     print("Reading the file with astropy.")
     t = Table.read(filename, format='ascii.csv', fast_reader=True)

